test/test_utils.py

```diff
@@ -268,3 +293,42 @@ def test_refused_response_no_question(self):
         r.set_rcode(dns.rcode.REFUSED)
         r.question = []
         utils.dnsmsg2log(r)
+
+
+class TestSSLContext(unittest.TestCase):
+    def setUp(self):
+        self._CA = TEST_CA
+        self._CA_serial = "E198832A55D5B708"
+
+    @unittest.skipUnless(ssl.HAS_ALPN, "Requires >=openssl-1.0.2")
```

Comment:
So... I would avoid skipping the tests.... as there is value in running those tests even openssl does not suport ALPN or NPN.

One way to do this is to patch: `ssl. set_alpn_protocols` and `ssl. set_npn_protocols` within `setUp`

Using the 3rd method describe in https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop 
```
...     def setUp(self):
...         patcher = patch('package.module.Class')
...         self.MockClass = patcher.start()
...         self.addCleanup(patcher.stop)
```

would apply those patches for each tests in `TestSSLContext`